src/node/create-webpack-config-client.js

```diff
@@ -123,6 +123,8 @@ function createWebpackConfigClient(
           name: 'manifest'
         },
         splitChunks: {
+          maxInitialRequests: 10,
+          maxAsyncRequests: 10,
```

Comment:
@katydecorah these changes also have implications for the client running the bundle (If I understand it correctly, the client will now have an upper limit of 10 parallel requests). Did you see any performance deterioration in environments like 3g connection or lower end mobile devices ?

Also, feel free to ignore my comment, as I am just trying to think of the worst case scenarios this could affect end users.

Reply:
@kepta Thanks for pointing this out. I should have noted in my original comment that there could be side effects to browsers/devices that do not support https/2. Based on a couple WebPageTest there is a slight deterioration on performance for these devices. (I checked our analytics and this would affect a very small fraction of visitors.)

 > I fear that these changes are driven by shortcoming of the build environment (publisher) and not the client, but it directly affects the client.

I agree, I would rather put the load on the build time rather than the client. We hit a sticking point, though, with mapbox-gl-js-docs running out of memory during build. 

Here are a couple of other points we have been thinking about:

* Since the build is running out of memory during minification, I am watching https://github.com/webpack-contrib/terser-webpack-plugin/issues/143 since it could be related. I confirmed that a bottleneck happens during minification by setting `minimize: false`: mapbox-gl-js-docs built without running out of memory and in a fraction of the time.
* We could increase the memory via `--max-old-space-size` to the build script, which would provide mapbox-gl-js-docs enough memory to build. However, this doesn't solve the root problem, but could buy us time until we are able to fix the root problem.

Would you happen to have any suggestions or recommendations on how we can prevent sites from running out of memory on build?

Reply:
> I agree, I would rather put the load on the build time rather than the client. 

Great! I think we should temporarily increase the heap size and continue investigating the root cause. If it is a bug on their end, maybe it gets fixed doing a version upgrade ?

> Would you happen to have any suggestions or recommendations on how we can prevent sites from running out of memory on build?

I can look into it if you can help me with the commit which is bloating the memory.

Reply:
Thanks so much @kushan, your guidance makes a lot of sense. 

* The commit in Batfish which introduced the memory bloat likely happened in https://github.com/mapbox/batfish/commit/717226a2686ad2f65803af28bb6d52094b2d8577 with the initial upgrade to Webpack@4. 
* The build runs out of memory at the `Bundling files for the browser ...` step.

Here's a progression of the commits and Batfish versions made to mapbox-gl-docs:

## Batfish@1.9.8

This Batfish version uses Webpack@3, I'm providing the latest build on this version in case it's helpful to compare the logs.

* Commit: https://github.com/mapbox/mapbox-gl-js-docs/commit/3b0395d6bebbaf9336f00403a83c07d1f3b3a956
* ✅ Build log: https://hey.mapbox.com/log-link/?logGroup=/aws/codebuild/artifacts-production-publisher&logStream=bda69ba7-2913-451f-9246-2cf97a5d75de#L527

## Batfish@1.10.2

This Batfish version upgrades to Webpack@4 (plus bug fixes).

* Commit: https://github.com/mapbox/mapbox-gl-js-docs/pull/60/commits/a149c15ab901700951c680cee8ed8d63ed87122e
* ❌Build log: https://hey.mapbox.com/log-link/?logGroup=/aws/codebuild/artifacts-production-publisher&logStream=6266e924-e909-4178-9e8f-a0a5e992f807#L528

(When debugging with node --inspect, the code had pointed at writing the stats.json file as running out of memory. So we shipped #307 in Batfish@1.11.0 to place stats.json behind a flag. However, I'm seeing in this build log that it was also pointing at terser.)



## Batfish@1.11.0

This Batfish version places the generation of stats.json behind a flag.

* Commit: https://github.com/mapbox/mapbox-gl-js-docs/pull/60/commits/5ca5149c38bd3b7bbbefda15d298c5367e6e0b5f
* ❌Build log: https://hey.mapbox.com/log-link/?logGroup=/aws/codebuild/artifacts-production-publisher&logStream=cc912089-c771-42c3-962b-ed0323338fde#L528

## Batfish@1.11.0 + increased Node memory

* Commit: https://github.com/mapbox/mapbox-gl-js-docs/commit/458d392281682c0db39515224e53a466e467a03b
* ✅Build log: https://hey.mapbox.com/log-link/?logGroup=/aws/codebuild/artifacts-production-publisher&logStream=ebde70c8-66cf-4033-b753-2963386c776c#L529


## Batfish@1.11.1

This Batfish version increases maxAsyncRequests and maxInitialRequests.

* Commit: https://github.com/mapbox/mapbox-gl-js-docs/pull/60/commits/95241238108308794325ff8d06a56c06ad8590fe
* ✅Build log: https://hey.mapbox.com/log-link/?logGroup=/aws/codebuild/artifacts-production-publisher&logStream=c750aa26-bdf2-46da-ba9a-dce90e9b22c6#L529